starport/services/network/queries.go

```diff
@@ -2,45 +2,76 @@ package network
 
 import (
 	"context"
+	"sync"
 
 	"github.com/pkg/errors"
 	campaigntypes "github.com/tendermint/spn/x/campaign/types"
 	launchtypes "github.com/tendermint/spn/x/launch/types"
+	rewardtypes "github.com/tendermint/spn/x/reward/types"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 
 	"github.com/tendermint/starport/starport/pkg/events"
 	"github.com/tendermint/starport/starport/services/network/networktypes"
 )
 
+const errNotFoundMsg = "rpc error: code = InvalidArgument desc = not found: invalid request"
```

Comment:
Can't we use another way to detect not found errors. Programmatically by using SDK or gRPC APIs.

I feel like it won't be reliable to depend on error strings.

Reply:
after merging this one https://github.com/tendermint/spn/pull/629, we can parse better the error codes from the SPN